go/service/teamsearch.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+// RPC handlers for team search operations
+
+package service
+
+import (
+	"context"
+
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/client/go/teams"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+)
+
+type TeamSearchHandler struct {
+	*BaseHandler
+	libkb.Contextified
+}
+
+var _ keybase1.TeamsInterface = (*TeamsHandler)(nil)
+
+func newTeamSearchHandler(xp rpc.Transporter, g *libkb.GlobalContext) *TeamSearchHandler {
+	return &TeamSearchHandler{
+		BaseHandler:  NewBaseHandler(g, xp),
+		Contextified: libkb.NewContextified(g),
+	}
+}
+
+func (h *TeamSearchHandler) TeamSearch(ctx context.Context, arg keybase1.TeamSearchArg) (res keybase1.TeamSearchRes, err error) {
+	ctx = libkb.WithLogTag(ctx, "TS")
+	if err := assertLoggedIn(ctx, h.G()); err != nil {
```

Comment:
Why not allow an unauthed user?

Reply:
The server side doesn't seem to allow it.  I don't really understand why, maybe I missed some flag on that side (though I just copied the featured bot stuff).  Since the server doesn't allow it, I did this here as well.

Another reason is that nojima wants an `inTeam` boolean on each team search result, but I guess that can always be false in the logged-out case.

Anyway I don't care much, happy to remove it if you'd prefer to see the server error instead.

For the record, `featured_bots/search` gives the same error:
```sh
$ keybase apicall -m GET -a query=test -a limit=1 teamsearch/search
▶ WARNING Running in devel mode
▶ ERROR Login required: no sessionArgs available since no login path worked
$ keybase apicall -m GET -a query=test -a limit=1 featured_bots/search
▶ WARNING Running in devel mode
▶ ERROR Login required: no sessionArgs available since no login path worked
```

Reply:
An unauthed user is never making it this far, no sense in worrying about it